tests/pipeline_backend_test.py

```diff
@@ -417,6 +423,43 @@ def tearDownClass(cls):
         cls.sc.stop()
 
 
+class TestLazySingleton(unittest.TestCase):
```

Comment:
should we maybe test it through public API? LazySingleton is implementaiton details

Reply:
Yeah, it's an implementation detail, but pretty non-trivial, so I'd prefer to test it.